panic when stop the raftstore

<!-- Please describe your issue in English. -->

#### Can be reproduced ?
Yes

#### Steps:
Run `make test` many times

#### Expected behavior:
PAAS

#### Actual behavior:
panic

#### Environment:
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:

#### Configuration file:


#### Additional context:
- Error message from client:
- Server log:
- Other information:

2021/08/31 23:43:38.226506 [info] [raftstore]: store 1 begin to stop
2021/08/31 23:43:38.226758 [error] prophet: client read resp from leader 127.0.0.1:30000 failed with read tcp 127.0.0.1:62413->127.0.0.1:30000: use of closed network connection
2021/08/31 23:43:38.226767 [error] prophet: client read loop stopped
2021/08/31 23:43:38.226826 [error] prophet: session 9[127.0.0.1:62452] read failed with read tcp 127.0.0.1:30000->127.0.0.1:62452: use of closed network connection
2021/08/31 23:43:38.226880 [error] prophet: watcher read events failed with EOF
2021/08/31 23:43:38.226888 [error] prophet: session 4[127.0.0.1:62412] read failed with read tcp 127.0.0.1:30000->127.0.0.1:62412: use of closed network connection
2021/08/31 23:43:38.226889 [error] prophet: reset watcher conn failed with <nil>, retry later
2021/08/31 23:43:38.226939 [error] prophet: session 7[127.0.0.1:62416] read failed with read tcp 127.0.0.1:30000->127.0.0.1:62416: use of closed network connection
2021/08/31 23:43:38.226959 [error] prophet: watcher read events failed with EOF
2021/08/31 23:43:38.226966 [error] prophet: reset watcher conn failed with <nil>, retry later
2021/08/31 23:43:38.226978 [error] prophet: client read resp from leader 127.0.0.1:30000 failed with EOF
2021/08/31 23:43:38.227011 [error] prophet: session 8[127.0.0.1:62451] read failed with read tcp 127.0.0.1:30000->127.0.0.1:62451: use of closed network connection
2021/08/31 23:43:38.227024 [info] prophet: client start init connection to leader 127.0.0.1:30000
panic: runtime error: index out of range [0] with length 0

goroutine 21261 [running]:
github.com/fagongzi/goetty/buf.Int2BytesTo(...)
	/Users/lni/go/pkg/mod/github.com/fagongzi/goetty@v1.8.0/buf/buf.go:65
github.com/fagongzi/goetty/codec/length.(*lengthCodec).Encode(0x14008139500, {0x103d907a0, 0x140015d8d80}, 0x140081db4f0)
	/Users/lni/go/pkg/mod/github.com/fagongzi/goetty@v1.8.0/codec/length/codec_length_field.go:89 +0x158
github.com/fagongzi/goetty.(*baseIO).write(0x14007d314a0, {0x103d907a0, 0x140015d8d80}, 0x0)
	/Users/lni/go/pkg/mod/github.com/fagongzi/goetty@v1.8.0/session.go:298 +0x70
github.com/fagongzi/goetty.(*baseIO).writeLoop(0x14007d314a0, {0x103dfc278, 0x14004ce3b00})
	/Users/lni/go/pkg/mod/github.com/fagongzi/goetty@v1.8.0/session.go:329 +0x150
created by github.com/fagongzi/goetty.(*baseIO).initConn
	/Users/lni/go/pkg/mod/github.com/fagongzi/goetty@v1.8.0/session.go:383 +0x218
FAIL	github.com/matrixorigin/matrixcube/raftstore	176.267s

--- a/components/prophet/option.go
+++ b/components/prophet/option.go
@@ -20,7 +20,7 @@ import (
 	"github.com/fagongzi/goetty/buf"
 	"github.com/matrixorigin/matrixcube/components/prophet/codec"
 	"github.com/matrixorigin/matrixcube/components/prophet/pb/metapb"
-	"github.com/matrixorigin/matrixcube/components/prophet/util"
+	"go.uber.org/zap"
 )
 
 // Option client option
@@ -54,7 +54,7 @@ func WithRPCTimeout(value time.Duration) Option {
 func createConn() goetty.IOSession {
 	encoder, decoder := codec.NewClientCodec(10 * buf.MB)
 	return goetty.NewIOSession(goetty.WithCodec(encoder, decoder),
-		goetty.WithLogger(util.GetLogger()),
+		goetty.WithLogger(zap.L().Named("cube-prophet-client")),
 		goetty.WithEnableAsyncWrite(16))
 
 }

--- a/components/prophet/prophet_transport.go
+++ b/components/prophet/prophet_transport.go
@@ -18,6 +18,7 @@ import (
 	"github.com/fagongzi/goetty/buf"
 	"github.com/matrixorigin/matrixcube/components/prophet/codec"
 	"github.com/matrixorigin/matrixcube/components/prophet/util"
+	"go.uber.org/zap"
 )
 
 func (p *defaultProphet) startListen() {
@@ -26,7 +27,7 @@ func (p *defaultProphet) startListen() {
 		p.handleRPCRequest,
 		goetty.WithAppSessionOptions(goetty.WithCodec(encoder, decoder),
 			goetty.WithEnableAsyncWrite(16),
-			goetty.WithLogger(util.GetLogger())))
+			goetty.WithLogger(zap.L().Named("cube-prophet-rpc"))))
 	if err != nil {
 		util.GetLogger().Fatalf("start transport failed with %+v", err)
 	}

--- a/go.mod
+++ b/go.mod
@@ -11,7 +11,7 @@ require (
 	github.com/cockroachdb/pebble v0.0.0-20210503173641-1387689d3d7c
 	github.com/coreos/go-semver v0.3.0
 	github.com/docker/go-units v0.4.0
-	github.com/fagongzi/goetty v1.8.0
+	github.com/fagongzi/goetty v1.10.1
 	github.com/fagongzi/log v0.0.0-20191122063922-293b75312445
 	github.com/fagongzi/util v0.0.0-20210409031311-a10fdf8fbd7a
 	github.com/go-ole/go-ole v1.2.4 // indirect
@@ -30,6 +30,7 @@ require (
 	go.etcd.io/etcd/client/v3 v3.5.0
 	go.etcd.io/etcd/raft/v3 v3.5.0
 	go.etcd.io/etcd/server/v3 v3.5.0
+	go.uber.org/zap v1.18.1
 	golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba
 )
 

--- a/raftstore/rpc.go
+++ b/raftstore/rpc.go
@@ -20,6 +20,7 @@ import (
 	"github.com/fagongzi/goetty/codec/length"
 	"github.com/matrixorigin/matrixcube/pb"
 	"github.com/matrixorigin/matrixcube/pb/raftcmdpb"
+	"go.uber.org/zap"
 )
 
 type defaultRPC struct {
@@ -36,7 +37,7 @@ func newRPC(store *store) *defaultRPC {
 	app, err := goetty.NewTCPApplication(store.cfg.ClientAddr, rpc.onMessage,
 		goetty.WithAppSessionOptions(goetty.WithCodec(encoder, decoder),
 			goetty.WithEnableAsyncWrite(16),
-			goetty.WithLogger(logger),
+			goetty.WithLogger(zap.L().Named("raftstore-rpc")),
 			goetty.WithReleaseMsgFunc(releaseResponse)))
 	if err != nil {
 		logger.Fatalf("create rpc failed with %+v", err)

--- a/server/server.go
+++ b/server/server.go
@@ -29,6 +29,7 @@ import (
 	"github.com/matrixorigin/matrixcube/pb/raftcmdpb"
 	"github.com/matrixorigin/matrixcube/proxy"
 	"github.com/matrixorigin/matrixcube/util"
+	"go.uber.org/zap"
 )
 
 var (
@@ -61,7 +62,7 @@ func NewApplicationWithDispatcher(cfg Cfg, dispatcher func(req *raftcmdpb.Reques
 		app, err := goetty.NewTCPApplication(cfg.Addr, s.onMessage,
 			goetty.WithAppSessionOptions(goetty.WithCodec(encoder, decoder),
 				goetty.WithEnableAsyncWrite(16),
-				goetty.WithLogger(logger),
+				goetty.WithLogger(zap.L().Named("cube-app")),
 				goetty.WithReleaseMsgFunc(s.releaseResponse)))
 		if err != nil {
 			logger.Fatalf("create internal tcp server failed with %+v", err)

--- a/transport/transport.go
+++ b/transport/transport.go
@@ -19,6 +19,7 @@ import (
 	"github.com/matrixorigin/matrixcube/pb/bhraftpb"
 	"github.com/matrixorigin/matrixcube/snapshot"
 	"go.etcd.io/etcd/raft/v3/raftpb"
+	"go.uber.org/zap"
 )
 
 var (
@@ -93,7 +94,7 @@ func NewDefaultTransport(
 	app, err := goetty.NewTCPApplication(addr, t.onMessage,
 		goetty.WithAppSessionOptions(goetty.WithCodec(t.encoder, t.decoder),
 			goetty.WithTimeout(t.opts.readTimeout, t.opts.writeTimeout),
-			goetty.WithLogger(logger),
+			goetty.WithLogger(zap.L().Named("cube-trans")),
 			goetty.WithEnableAsyncWrite(t.opts.sendBatch)))
 	if err != nil {
 		logger.Fatalf("create transport failed with %+v", err)

